runtime/src/bank.rs

```diff
@@ -1311,6 +1327,47 @@ impl Bank {
         self.update_recent_blockhashes_locked(&blockhash_queue);
     }
 
+    fn get_timestamp_estimate(&self) -> Option<UnixTimestamp> {
+        let mut get_timestamp_estimate_time = Measure::start("get_timestamp_estimate");
+        let recent_timestamps: HashMap<Pubkey, (Slot, UnixTimestamp)> = self
+            .vote_accounts()
```

Comment:
Is this the rooted vote accounts? That is, are we sure all validators will calculate the same timestamp?

Reply:
These are the vote accounts in the new (from parent) Bank. Same info that is used for rewards, epoch-stakes, etc. I think that if we have nodes coming up with different Bank/accounts state for the same slot, we have a bigger problem?